File: leet_TREE_0226_mirror__post.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(node *TreeNode) *TreeNode {
    if node == nil {
        return node
    }
    invertTree(node.Left)
    invertTree(node.Right)
    node.Left, node.Right = node.Right, node.Left
    return node
}

File: leet_TREE_0226_mirror__post.cc

class Solution {
public:
    TreeNode* invertTree(TreeNode* node) {
        if (!node)
            return NULL ;
        invertTree(node->right);
        invertTree(node->left);
        swap( node->left, node->right );
        return node ;
    }
};

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
